lib/cdo/test_flakiness.rb

```diff
@@ -44,7 +44,12 @@ def self.get_jobs(options = {})
   def self.summarize_by_job(num_requests = NUM_REQUESTS, per_request = PER_REQUEST)
     jobs = []
     num_requests.times do
-      jobs += get_jobs(limit: per_request, skip: jobs.count)
+      options = {
+        limit: per_request,
+        skip: jobs.count,
+        from: from_timestamp
```

Comment:
The `from` parameter isn't very well documented in the [SauceLabs API docs](https://wiki.saucelabs.com/display/DOCS/Job+Methods).

```
from (optional timestamp)
Get jobs since the specified time
```

What have you learned about the behavior of this parameter?  Does it affect the ordering or pagination of results at all?  I've thought of two scenarios that might have issues.

### Scenario one: from is a filter

If ordering and pagination works the same way, but `from` filters out results before a given timestamp, our code might make more requests than it needs to.  Suppose we just reset flakiness an hour ago, only a hundred jobs have run since then, and we're running `summarize_by_job` with its default parameters.

In this scenario, our first request would return 100 results.  Then we'd make 49 more requests giving no results, which we could probably skip.  This wouldn't break our results, but it might be nice to optimize these out.

### Scenario two: from is a starting point

Another (less-likely but) possible interpretation of this API is that `from`, when provided, acts as a starting point. (i.e. "I want 100 jobs starting `from` this timestamp.")  In that scenario, we would start getting incorrect results once our `from` value passed out of our existing flakiness window.  If we make 50 requests and get 500 results each, and they started at `from`, then 25,000 jobs later our flakiness stats would stop changing because we would be retrieving the same jobs every time.

Reply:
fortunately, scenario 1 is true, but thank you for making me double-check 😅   i confirmed this by running `bin/test_flakiness` with and without a timestamp, then checking that the timestamp run is a subset of the timestamp-less run results.

re:
> If ordering and pagination works the same way, but from filters out results before a given timestamp, our code might make more requests than it needs to.

this was also true, so i added [this commit](https://github.com/code-dot-org/code-dot-org/pull/36051/commits/61b29a71bbc943a3573d9fe818672fc4a635141a), which breaks if we request jobs and get an empty result